lib/generators/rspec/integration/integration_generator.rb

```diff
@@ -10,12 +10,58 @@ class IntegrationGenerator < Base
                    :type => :boolean,
                    :default => true,
                    :desc => "Generate request specs"
+      class_option :api, :type => :boolean, :desc => "Creates request_spec for APIs, skip specs unnecessary for API-only apps"
+      class_option :fabrication, :type => :boolean, :desc => "Fill params with Fabricator model attributes"
+      class_option :factorybot, :type => :boolean, :desc => "Fill params with Factory Bot model attributes"
+
+
+      def initialize(*args, &blk)
+        @generator_args = args.first
+        super(*args, &blk)
+      end
 
       def generate_request_spec
         return unless options[:request_specs]
+        
+        if options[:api]
+          template 'api_request_spec.rb', File.join('spec/requests', class_path, "#{table_name}_spec.rb")
+        else
+          template 'request_spec.rb',
+                   File.join('spec/requests', class_path, "#{table_name}_spec.rb")
+        end
+
+      end
+
+      protected
+
+      attr_reader :generator_args
+
+      # @todo refactor the following methods. They are also in the /lib/generators/rpsec/scaffold/scaffold_generator.rb file.
```

Comment:
Either refactor these, or remove the todo. If this PR isn't complete it can't be merged.